File: leetcode75/05-345. Reverse Vowels of a String/main.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowel_set = ("a", "A", "e", "E", "i", "I", "o", "O", "u", "U")
        stack = []
        for c in s:
            if c in vowel_set:
                stack.append(c)

        result = ""
        for c in s:
            if c in vowel_set:
                result = result + stack.pop()
            else:
                result = result + c
        return result


s = Solution()
print(s.reverseVowels("IceCreAm"))

File: leetcode75/05-345. Reverse Vowels of a String/main.rs
use std::{collections::HashSet, hash::Hash, result};

    pub fn reverse_vowels(s: String) -> String {
        //let vowel_set = HashSet::new()
        // vowel_set.insert("a")
        let vower_set: HashSet<char> = HashSet::from(['a', 'A', 'e', 'E', 'i', 'I', 'o', 'O', 'u', 'U']);
        let mut stack = vec![];
        for c in s.chars() {
            if vower_set.contains(&c) {
                stack.push(c);
            }
        }

        let mut result = String::new();
        for c in s.chars(){
            if vower_set.contains(&c) {
                result.push(stack.pop().unwrap());
            } else {
                result.push(c);
            }
        }
        return result;
    }


fn main() {
    println!("Reverse {}",reverse_vowels(String::from("IceCreAm")))
}
